### DISCUSSION TITLE
help solving 'drawBmp' was not declared in this scope

### DISCUSSION THREAD

**Discussion by bmarcu13** [Bodmer/TFT_eSPI#992] [Q&A - General]
I have the following code but it keeps giving me this error: 'drawBmp' was not declared in this scope.
Any help would be appreciated.
Thanks. 

```
#define FS_NO_GLOBALS
#include <FS.h>
#include <TFT_eSPI.h>

#include <ESP8266WiFi.h>
#include <ESP8266WebServer.h>
#include <Firebase_ESP_Client.h>

int ok = 0;

TFT_eSPI tft = TFT_eSPI();

void setup() {
  Serial.begin(9600);

  if (!SPIFFS.begin()) {
    Serial.println("SPIFFS initialisation failed!");
    while (1) yield();
  }
  Serial.println("\r\nSPIFFS initialised.");
  
  tft.begin();
  tft.fillScreen(TFT_BLACK);

  WiFi.begin(WIFISSID, WIFIPassword);
  while (WiFi.status() != WL_CONNECTED)
  {
    delay(500);
    Serial.print(".");
  }

  config.host = FIREBASE_HOST;
  config.signer.tokens.legacy_token = FIREBASE_AUTH;

  Firebase.begin(&config, &auth);
  Firebase.reconnectWiFi(true);
  fbdo.setBSSLBufferSize(1024, 1024);
  fbdo.setResponseSize(1024);
}

void loop() {
  delay(5000);
  if (ok == 0) {
     if (Firebase.Storage.download(&fbdo, STORAGE_BUCKET_ID, "heart1.bmp", "/heart1.bmp", mem_storage_type_flash))
    {
        Serial.println("PASSED");
        Serial.println("------------------------------------");
        Serial.println();
        ok = 1;
    }
    else
    {
        Serial.println("FAILED");
        Serial.println("REASON: " + fbdo.errorReason());
        Serial.println("------------------------------------");
        Serial.println();
    }
  }
  else if(ok == 1){
      int x = 10;
      int y = 10;
      drawBmp("/heart1.bmp", x, y);
      ok = 2;
  }
  Serial.println(ok);
```

**Comment by Bodmer** [comment]
drawBmp is not a function in the library, it needs to be in your sketch.

Experiment with [this example](https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/TFT_SPIFFS_BMP).

**Reply by bmarcu13** [reply]
Now it gives me 
`BMP format not recognized.`

What format does the file must have to be compatible with the library? (I use a .bmp file exported from Photoshop)

**Comment by Bodmer** [comment]
The BMP file must be in 24 bit colour, uncompressed format. This is the default for most image creation software like MS Paint.

**Reply by bmarcu13** [reply]
Thanks for the reply. It works now according to the logs (it says it is loading the bitmap successfully now) but is seems like it isn't recognizing my display since it boots with white backlight.

Here is the code I'm using for the TFT_eSPI library:

What I find strange is that on the Serial monitor is never printed the "nSPIFFS initialised." message, it jumps directly to the WiFi instructions which are displayed on the Serial Monitor.

Am I not doing the initialization right?

PS. 
-The pins are correctly defined since they are the same I used to be using with the Adafruit Library (D4 -TFT_DC & D1 - TFT_CS)
-I actually have a ILI1340 display but on the Adafruit website it says the two models are almost identical and they can use the same library. 

```
#include <TFT_eSPI.h>
#include <SPI.h>

#define ILI9341_DRIVER

#define TFT_DC D4
#define TFT_CS D1

#define LOAD_GLCD

#define SPI_FREQUENCY  27000000
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000

TFT_eSPI tft = TFT_eSPI();

void setup() {
  Serial.begin(9600);

  if (!SPIFFS.begin()) {
    Serial.println("SPIFFS initialisation failed!");
    while (1) yield(); // Stay here twiddling thumbs waiting
  }
  Serial.println("\r\nSPIFFS initialised.");
  
  tft.begin();
  tft.fillScreen(TFT_BLACK);
  tft.setRotation(0);

  WiFi.begin(WIFISSID, WIFIPassword);
  while (WiFi.status() != WL_CONNECTED)
  {
    delay(500);
    Serial.print(".");
  }
}
```

**Comment by Bodmer** [comment]
The #define's in the sketch are not seen by the library so have no effect on the configuration. You have to modify the setup file.

Run the "Read_User_Setup" example and look at the Serial Monitor window output to see what setting the compiler is picking up, check the output is as you expect.  If it is wrong then you have not configured the library correctly. [Read this.](https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h)

**Reply by bmarcu13** [reply]
I managed to make it work by editing the "User_Setup_Select".
But when I try to load [this](https://imgur.com/a/ALYELJO) bitmap that I have created in MS Paint it only shows the white background of it, like [this](https://imgur.com/a/9hkhA3O)

Also, when running  "Read_User_Setup" and opening the Serial Monitor, this shows up on it:
```
09:03:18.525 -> ⸮⸮ 1⸮O⸮
09:03:18.525 -> &⸮! ⸮H1	⸮!⸮k	H⸮
```

**Reply by Bodmer** [reply]
Read_User_Setup uses a baud rate of 115200, make sure your terminal window uses the same rate.
